lib/after.js

```diff
@@ -33,33 +33,29 @@ test.afterEach( function() {
 
 	const screenSize = driverManager.currentScreenSize().toUpperCase();
 	const locale = driverManager.currentLocale().toUpperCase();
+	const date = new Date().getTime().toString();
 
 	if ( this.currentTest.state === 'failed' ) {
-		const prefix = `FAILED-${locale}-${screenSize}-${shortTestFileName}`;
-
 		let neverSaveScreenshots = config.get( 'neverSaveScreenshots' );
 		if ( neverSaveScreenshots ) {
 			return null;
 		}
 
-		driver.getCurrentUrl().then( ( url ) => console.log( `FAILED: Taking screenshot of: '${url}'` ), ( err ) => {
+		driver.getCurrentUrl().then( url => console.log( `FAILED: Taking screenshot of: '${url}'` ), ( err ) => {
 			slackNotifier.warn( `Could not capture the URL when taking a screenshot: '${err}'` );
 		} );
 
-		return driver.takeScreenshot().then( ( data ) => {
-			mediaHelper.writeScreenshot( data, prefix );
-		}, ( err ) => {
-			slackNotifier.warn( `Could not take screenshot due to error: '${err}'` );
-		} );
-	}
-	if ( config.get( 'saveAllScreenshots' ) === true ) {
-		const prefix = [ `${locale}-${screenSize}-${shortDescribeFileName}`, `${shortTestFileName}` ];
-		return driver.takeScreenshot().then( ( data ) => {
-			mediaHelper.writeScreenshot( data, prefix, true );
-		}, ( err ) => {
-			slackNotifier.warn( `Could not take screenshot due to error: '${err}'` );
-		} );
+		filenameCallback = () => `FAILED-${locale}-${screenSize}-${shortTestFileName}-${date}`;
```

Comment:
Since this variable is never defined with `let`, `var`, or `const`, I think it becomes a global variable, which is probably not what you want.

Reply:
Indeed, thanks! (I just quickly pushed this WIP file up for review)